pkg/websockets/socket.go

```diff
@@ -0,0 +1,251 @@
+package websockets
+
+import (
+	"errors"
+	"log"
+	"time"
+
+	"github.com/codeready-toolchain/registration-service/pkg/middleware"
+	"github.com/gin-gonic/gin"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	// time allowed to write a message to the peer.
+	writeWait = 10 * time.Second
+	// time allowed to read the next pong message from the peer.
+	pongWait = 10 * time.Second
+	// send pings to peer with this period. Must be less than pongWait.
+	pingPeriod = (pongWait * 9) / 10
+	// maximum message size allowed from peer.
+	maxMessageSize = 512
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+}
+
+// Message represents a message over the channel
+type Message struct {
+	// subscription of the peer identity
+	Sub string
+	// message data
+	Body []byte
+}
+
+// Hub maintains the set of active clients
+type Hub struct {
+	// registered clients, mapped to the sub id.
+	clients map[*Client]string
```

Comment:
If I got it right you want to use the same Hub for multiple ws connections. So if there is a change in any Signup on the server side then we will notify the corresponding connection/client if there are any registered for that signup/userID.

So, on the server side we will probably use `oc get signups -w` kinda thing. To watch all signups in the K8s API. And if there any changes we write/transfer it to the hub and the hub broadcast it to the corresponding clients.

Is it correct?